### DISCUSSION TITLE
[Application] File.app

### DISCUSSION THREAD

**Discussion by jimmylee** [filecoin-project/community#109] [Project Showcase]
Learn about https://file.app, built by Applications Research Group (ARG), [here](https://file.app).

Team:
@jimmylee

**Comment by jimmylee** [comment]
![social](https://user-images.githubusercontent.com/310223/111859123-a74de180-88fb-11eb-89bb-482f71f8095d.png)

### Updates

For the week ending on Friday, Mar 19th

- We launched https://file.app
- It is one of the easiest ways to find out how cheap Filecoin mining is and also the overall performance of the Filecoin network and the price of Filecoin!
- We use IEXCloud for the Filecoin Price! FIL/USDT
- We use https://space-race-slingshot-phase2.s3.amazonaws.com/prod/unfiltered_basic_stats.json for Filecoin Network Stats
- We use https://filrep.io/api for one miner index
- We use https://bafzbeidtcvgkyhof2jqy35xyevtxefqkfubkrh7lvrenn53uz3srea3rcu.textile.space/ for the other miner index
- I tweeted about it here: https://twitter.com/wwwjim/status/1373129266880147461?s=20

**Comment by jimmylee** [comment]
<img width="1946" alt="Screen Shot 2021-03-27 at 12 43 57 AM" src="https://user-images.githubusercontent.com/310223/112714073-c7e6e000-8e95-11eb-86bc-bfd5178ffb71.png">
 
### Updates

For the week ending on Friday, Mar 26th, 2021

- added percentages for how much cheaper Filecoin is than Amazon S3's infrequent usage tier for @protocolin.
- provided conversion from Amazon's $0.0125 per GB per month ➟ to $0.01342 per GiB per month.
- provided a reference key `GiB`, `MiB`, `KIB`, and `Fil-Epoch` so people understand what that even means.
- Explained the way we calculate these percentage differences

```
METHOD ➟ ((AMAZON COST - FILECOIN COST) / FILECOIN COST) * 100
EXAMPLE MONTHLY (STALE NUMBERS) ➟ (($0.0134217728 - $0.0000527) / $0.0000534) * 100 = 25373.61%
```

- **1 GiB per day** - 25404.0046228866% cheaper
- **1 GiB per month (30 days)** - 25404.0046228866% cheaper
- **1 GiB per year** - 25054.634696545687% cheaper

**Reply by moonbread** [reply]
@jimmylee neat page! Thanks for sharing! Do you know if historical storage stats and total network storage power data is available anywhere? Looking to see how it has evolved since mainnet launch.

**Reply by jimmylee** [reply]
I don't know and don't want to misinform... but I think @andrewxhill or @smooth-operator might have some ideas of how to query their APIs for it?

**Reply by andrewxhill** [reply]
we don't have an api for that history... no. some chain wizardry might be able to pull it out though.

**Reply by jimmylee** [reply]
Yeah if anyone has any ideas of how to help, would be extremely useful.

**Comment by jimmylee** [comment]
<img width="1750" alt="Screen Shot 2021-04-02 at 5 10 12 PM" src="https://user-images.githubusercontent.com/310223/113462573-6dd6a500-93d6-11eb-89e1-e7eb9c1b9293.png">

## Updates

For the week ending on Friday, April 2nd, 2021

How we calculate price is now very clear

```
➟ For each of the 439 filtered miners
➟ take the "average amount spent on deals" from each miner
➟ divide by "total data in bytes stored by 439 miners" 

obtain the "attoFIL per byte"

➟ then multiply by 1073741824 bytes to get "average attoFIL per GiB per FIL-epoch".
```

I also cleaned up our other calculations

```
➟ Amazon S3 - Infrequent Access tier costs $0.0134217728 per GiB per month.
➟ ((Amazon Cost - Filecoin Cost) / Filecoin Cost) * 100 ➟ to get times cheaper
➟ (($0.0134217728 - $0.0000527) / $0.0000527) * 100 = 25368.26% ➟ Filecoin is 25368.26% times cheaper.
``` 

I also added another way of framing the calculation, thank you @marten-seemann for the idea.

```
Alternate calculation 
➟ Filecoin cost / Amazon cost
➟ 0.0000522 / 0.0134217728 = 0.00388920307
```

**Filecoin is 0.38% the cost of Amazon S3 - Infrequent Access per GiB per month.**

I also changed the design based off of feedback I saw on Reddit. I didn't modernize it, but I definitely made it more boring and clear so more people could enjoy it <3

**Comment by jimmylee** [comment]
<img width="1965" alt="Screen Shot 2021-04-09 at 3 39 31 PM" src="https://user-images.githubusercontent.com/310223/114247394-d8915e80-9949-11eb-87cd-be66014929a1.png">

<img width="1965" alt="Screen Shot 2021-04-09 at 3 39 41 PM" src="https://user-images.githubusercontent.com/310223/114247404-dfb86c80-9949-11eb-93d5-83068aba80cc.png">

## Updates

For the week ending on Friday, April 9th, 2021

* From every single Miner you can visit a miner specific profile on https://storage.market to make a storage deal with that miner.
* A clean up on design based on feedback to make https://file.app easier to read and more shareable.

**Comment by jimmylee** [comment]
<img width="2154" alt="Screen Shot 2021-05-05 at 1 46 13 AM" src="https://user-images.githubusercontent.com/310223/117116863-ba99fc80-ad43-11eb-951d-eb47fa4b9ed5.png">

## Updates

For the week ending on May, 7th, 2021

* Updated the price comparison: **0.45% the cost of Amazon S3** so that people know Filecoin is 0.45% the cost of Amazon S3 and that is measured over at least 594,637 deals.
